### DISCUSSION TITLE
Uasyncio Timeout Error

### DISCUSSION THREAD

**Discussion by case485** [micropython/micropython#11684] [General]
Need some assistance with async, I wrote a small program to monitor ADC values and used async to check the voltage values but when ever I enable the part of the function to first connect to wifi the async errors out , if i comment out the wifi connect all is good. When enabled the error is triggered by an ADC.read() call.  Not sure if I am missing something obvious here. Thanks for any help. Great work on the project!

The error I get is : 
Traceback (most recent call last):
  File "<stdin>", line 7, in <module>
  File "uasyncio/core.py", line 1, in run
  File "uasyncio/core.py", line 1, in run_until_complete
  File "uasyncio/core.py", line 1, in run_until_complete
  File "<stdin>", line 4, in main
  File "uasyncio/funcs.py", line 1, in gather
  File "uasyncio/core.py", line 1, in run_until_complete
  File "<stdin>", line 4, in check_limit_switches
OSError: [Errno 116] ETIMEDOUT: ESP_ERR_TIMEOUT

Here is the code : 
    from machine import Pin, ADC
    from time import sleep
    import uasyncio
    import network

    def setup_adc_list(pin):
        gpio=Pin(pin, Pin.IN, Pin.PULL_UP)
        adcPin = ADC(gpio)
        adcPin.atten(ADC.ATTN_11DB)
        return(adcPin)

    def scanForWifiNetworks():
        sta_if = network.WLAN(network.STA_IF)
        sta_if.active(True)
        authmodes = ['Open', 'WEP', 'WPA-PSK' 'WPA2-PSK4', 'WPA/WPA2-PSK']
        for (ssid, bssid, channel, RSSI, authmode, hidden) in sta_if.scan():
          if ("Open" in authmodes[authmode]) and (ssid is not None):
              ssid = (ssid).decode("utf-8")
              print("Authmode open for:", ssid )
              return(ssid)
        print("No valid SSIDS found open")
        return(0)
    
    def connectToWifi(ssid = None):
        station = network.WLAN(network.STA_IF)
        if station.isconnected() == True:
            ipAddr, netmask, router, dns = station.ifconfig()
            print("Already connected with IP Address: ", ipAddr)
            return(ipAddr)
        if (ssid == None):
            ssid = scanForWifiNetworks()
        if(ssid != ""):
            station.active(True)
            print("Not currently connected to Wifi, attempting connection to : ", ssid)
            station.connect(ssid)
            while(station.isconnected() == False):
                pass            
            print("Connection Successful") 
            print(station.ifconfig())
            ipAddr, netmask, router, dns = station.ifconfig()
            return(ipAddr)
        else:
            print("Did not recieve open ssid to connect to.... Trying Again")
            connectToWifi()      

    async def check_limit_switches(pin, sleepTime=3):  
        MAX_DATA_LIST_SIZE = -10
        while True:
            val = pin.read()
            val = val * (3.3 / 4095)
            pinNum = str(pin).split(",")[0].strip("ADC(")
            pinNum = int(pinNum[pinNum.find("(")+1:pinNum.find(")")])
            await uasyncio.sleep(1)
            print(f"PIN: {pin} Value: {val}")
            gc.collect()

    async def main(adcPin):
        print(f"ADC LIST2: {adcPin}")
        taskGroup = await uasyncio.gather(
            check_limit_switches(adcPin)
        )

    if __name__ == "__main__":
        adcPinsFor12vMonitoring = 15
        adcPin = setup_adc_list(adcPinsFor12vMonitoring)
        print(f"ADC LIST: {adcPin}")
        ipAddr = connectToWifi("West3")  <--------------------------------------What causes the fault
       
        uasyncio.run(main(adcPin))

**Comment by bixb922** [comment]
What chip/board are you using? If it is of the ESP32 family, please check the Espressif documentation if pin 15 is free to use for ADC. For example, pin 15 of the ESP32 belongs to ADC block 2 which should not be used when WiFi is active (see https://docs.espressif.com/projects/esp-idf/en/v4.4/esp32/api-reference/peripherals/adc.html for the ESP32 for ADC limitations).

**Comment by peterhinch** [comment]
Note that `connectToWifi` is recursive with no limit to recursion depth. However it's clear that the ADC read is failing which is most unlikely to be anything to do with `uasyncio`. I think @bixb922 may have the answer.

**Comment by case485** [comment]
@bixb922 - You answered it. I am running a ESP32-DevKit1 v1 and by using the ADC on 35 and 39 the issues resolved. Thanks a ton for the help I would have never found that. 
![ESP32-DOIT-DEV-KIT-v1-pinout-mischianti](https://github.com/micropython/micropython/assets/17994779/d67d7b91-1fdd-43a5-b003-b7ffac819f55)
